pkg/plugin/aggregation/run.go

```diff
@@ -127,24 +127,31 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	go aggr.IngestResults(monitorCh)
 
 	// Ensure we only wait for results for a certain time
-	timeout := time.After(time.Duration(cfg.TimeoutSeconds) * time.Second)
+
+	// softTimeout will start by cleaning up all plugins, but we aren't in an error state yet.
+	softTimeout := time.After(time.Duration(cfg.TimeoutSeconds) * time.Second)
+	// hardTimeout means we've waited way too long and are in an error state now. Worst case scenario.
+	hardTimeout := time.After(time.Duration(cfg.TimeoutSeconds+300) * time.Second)
```

Comment:
open to suggestions on the `+300` magic number

Reply:
at least make it a constant